docs/develop/java/overview.mdx

```diff
@@ -4,79 +4,100 @@ description: "Get an idea of what a Restate Java service looks like."
+When using composite types/POJOs for input/output and state, [Jackson Databind](https://github.com/FasterXML/jackson) will be used by default.
+The Jackson dependency is not automatically included, you must add it with `sdk-serde-jackson`:
 
-Now you need to configure the protobuf plugin to build your Protobuf contracts.
-Have a look at the [`build.gradle.kts` file of the Java Gradle template](https://github.com/restatedev/examples/blob/main/templates/java-gradle/build.gradle.kts) for an example.
+```kotlin
+implementation("dev.restate:sdk-serde-jackson:VAR::JAVA_SDK_VERSION")
+```
 
-For Maven, you can use the [xolstice Protobuf plugin](https://www.xolstice.org/protobuf-maven-plugin/index.html).
+</TabItem>
+<TabItem value="kotlin" label="Kotlin">
 
-## API Examples
+Run:
 
-Below is an example of a Java service. 
+```shell
+gradle init --type kotlin-application
+```
 
-Restate uses [protobuf](https://protobuf.dev/) to define the service and its methods. The example shows a greeter service with a single method:
-- `greet`: maintains a record of the number of times it has received a request for a given name, and prints a greeting.
+Add the [Kotlin symbol processing](https://kotlinlang.org/docs/ksp-quickstart.html#use-your-own-processor-in-a-project) and the [Kotlin serialization](https://kotlinlang.org/docs/serialization.html) plugin:
 
-```java
-CODE_LOAD::java/src/main/java/develop/Greeter.java
+```kotlin
+plugins {
+    kotlin("plugin.serialization") version "1.9.22"
+    id("com.google.devtools.ksp") version "1.9.22-1.0.18"
+}
 ```
 
-The `Greeter` class extends `GreeterRestate.GreeterRestateImplBase`, which is the base generated by the Restate protoc generator.
-It has the name of the service (`Greeter`) with `Restate` appended to it.
+Add the runtime dependency `sdk-api-kotlin` and the ksp dependency `sdk-api-kotlin-gen`:
 
-The [Restate context](/develop/java/restate-context) is supplied to the method as the first argument.
-The context enables interaction with Restate (e.g. calling other services, setting timers).
-The `greet` method uses an ObjectContext since it is a keyed service. For unkeyed services, use `Context`.
-Finally, the Restate server is set up to serve both methods of the greeter service.
+```kotlin
+ksp("dev.restate:sdk-api-kotlin-gen:VAR::JAVA_SDK_VERSION")
+implementation("dev.restate:sdk-api-kotlin:VAR::JAVA_SDK_VERSION")
+```
 
-This is what corresponding Protobuf definition would look like for this service:
+</TabItem>
+</Tabs>
 
-```protobuf
-CODE_LOAD::java/src/main/proto/greeter.proto
-```
+Depending on the deployment target, add one of the following dependencies:
+
+- To run as long-running app: `dev.restate:sdk-http-vertx:VAR::JAVA_SDK_VERSION`
+- To run on AWS Lambda: `dev.restate:sdk-lambda:VAR::JAVA_SDK_VERSION`
 
-To understand the Restate-specific parts of this Protobuf definition, have a look at the documentation [here](/develop/service_type#restate-service-contract).
+## API Examples
```

Comment:
The use of virtual object vs service is confusing in this section.
It starts with "Below is an example of a virtual object." And ends with "Now that you have a high-level idea of what a Restate service might look like,". This is exactly why we need a different name for services... Not sure how to fix it. Let's discuss on Slack.